dashboard/frontend/src/pages/Job.tsx

```diff
@@ -142,6 +153,38 @@ const JobPage: FC<{
     await handler()
   }, [])
 
+  const loadInputRelationInfo = useCallback(async () => {
+    const handler = loadingErrorHandler(async () => {
+      const info = await api.get(`/api/v1/job/${id}/inputs`);
+      setJobInputRelation(info);
+    });
+    await handler();
+  }, []);
```

Comment:
We need to include `id` in the list of dependent values for this `useCallback` - otherwise it will memoize the old `id` value and bad bugs start to happen.

For context - for `useCallback` and `useMemo` functions - this array you pass is the list of memoized values that are used inside the function...  The idea is if any of those values change - the function is re-memoized but if none of those values have changed (between renders) - then re-use the old memoized value.

In the example above - we must include `id` in the list such that when the value of `id` changes - the function will re-memoize itself:

```ts
  const loadInputRelationInfo = useCallback(async () => {
    const handler = loadingErrorHandler(async () => {
      const info = await api.get(`/api/v1/job/${id}/inputs`);
      setJobInputRelation(info);
    });
    await handler();
  }, [
    id,
  ]);
```

A trick for doing this generally is to always read through the inner function and spot any variables that are defined outside the scope of that function (in this case `id`) and always just include those in the final array.

Reply:
NOTE - I've just realised the original code didn't have the `id` included in the list in which case this is why reviews are good :-)